### ISSUE TITLE
Finding out eigen values and vectors of huge sparse matrices in python is horribly slower than Matlab

### ISSUE THREAD

**Issue by SRITANU** [scipy/scipy#10192]
I am trying to find 100 eigen values and vectors of a huge sparse matrix of size 409600x409600. I am using `scipy.sparse.linalg.eigs` for this and it's taking ages to find the result whereas `eigs` on matlab solves it within 10 mins. Any suggestions on how to speed it up?

**Comment by ev-br** [comment]
Would be helpful to profile `sparse.eigs` to see if there are bottlenecks in the python level stuff. Runtime should be dominated by ARPACK, so the underlying implementation should be the main factor.

**Comment by scottwat86** [comment]
It may be throughout problem with ARPACK processing eigenvalues near 0. The linked documentation suggests performaning the computation in shift-invert mode.

https://docs.scipy.org/doc/scipy/reference/tutorial/arpack.html

 Thanks for the tip ev-br.

**Comment by bhourahine** [comment]
I just bumped into this post while trying to work out why scipy.sparse.linalg eigsh is showing scaling that is worse than quartic in the matrix order n, and finding the smallest eigenvalue of a simple tridagonal problem (the k-th eigenvalue for this specific matrix is  analytically located at k^2, so the 1st is well separated from the 2nd, but with a matrix condition number growing as n^2). I've used ARPACK with other languages, so wasn't expecting anything worse than cubic behavior here. This is for scipy 1.11.4 on Python 3.12.3.
